src/main/java/gregtech/common/covers/detector/CoverDetectorBase.java

```diff
@@ -0,0 +1,91 @@
+package gregtech.common.covers.detector;
+
+import codechicken.lib.raytracer.CuboidRayTraceResult;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.ICoverable;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.text.TextComponentTranslation;
+
+public abstract class CoverDetectorBase extends CoverBehavior {
+    protected static final String NBT_KEY_IS_INVERTED = "isInverted";
+
+    private boolean isInverted;
+
+    public CoverDetectorBase(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        isInverted = false;
+    }
+
+    protected boolean isInverted() {
+        return this.isInverted;
+    }
+
+    protected void setInverted(boolean isInverted) {
+        this.isInverted = isInverted;
+    }
+
+    private void toggleInvertedWithNotification() {
+        setInverted(!isInverted());
+
+        if (!this.coverHolder.getWorld().isRemote) {
+            this.coverHolder.writeCoverData(this, 100, b -> b.writeBoolean(isInverted()));
```

Comment:
Is there already or should there be an entry for this network id value in `GregTechDataCodes`?